src/fondant/pipeline/compiler.py

```diff
@@ -515,3 +517,167 @@ def _set_configuration(self, task, fondant_component_operation):
             task.set_accelerator_type(accelerator_name)
 
         return task
+
+
+class SagemakerCompiler(Compiler):
+    def __init__(self):
+        self._resolve_imports()
+
+    def _resolve_imports(self):
+        try:
+            import sagemaker
+            import sagemaker.processing
+            import sagemaker.workflow.pipeline
+            import sagemaker.workflow.steps
+
+            self.sagemaker = sagemaker
+
+        except ImportError:
+            msg = """You need to install the sagemaker extras to use the sagemaker compiler,\n
+                     you can install it with `pip install fondant[sagemaker]`"""
+            raise ImportError(
+                msg,
+            )
+
+    def build_command(
```

Comment:
So this could be reused by other compilers and be abstracted more

Reply:
would make it an internal function and rename it to `get_build_command()`